lib/info.js

```diff
@@ -179,19 +179,23 @@ const gotConfig = (id, options, additional, config, fromEmbed, callback) => {
       info.fmt_list.map((format) => format.split('/')) : [];
 
     info.formats = parseFormats(info);
+    try {
+      // Add additional properties to info.
+      Object.assign(info, additional, {
+        video_id: id,
 
-    // Add additional properties to info.
-    Object.assign(info, additional, {
-      video_id: id,
+        // Give the standard link to the video.
+        video_url: VIDEO_URL + id,
 
-      // Give the standard link to the video.
-      video_url: VIDEO_URL + id,
+        // Copy over a few props from `player_response.videoDetails`
+        // for bakcwards compatibility.
+        title: info.player_response.videoDetails.title,
+        length_seconds: info.player_response.videoDetails.lengthSeconds,
```

Comment:
would rather check here if `videoDetails` is defined, rather than wrap this whole thing in a `try` block. otherwise, private videos will not get the other properties that do not depend on `videoDetails`, such as `video_id` and `video_url`

Reply:
Thanks! I totally agree.

Reply:
+ the case which age-restricted couldn't downloadable was https://www.youtube.com/watch?v=bnb8-jpkQnc
It was `age-restricted` and `disallow embedding`.
sorry for mistake